cli/src/pcluster/models/imagebuilder.py

```diff
@@ -12,160 +12,112 @@
-            volume_type=ROOT_VOLUME_TYPE,
-            volume_size=self.image.root_volume.size,
-        )
-        self._execute_validator(
-            AMIVolumeSizeValidator,
-            volume_size=self.image.root_volume.size,
-            image=self.build.parent_image,
-        )
+import json
+import logging
+
+from common.aws.aws_api import AWSApi
+from common.aws.aws_resources import Stack
+from common.boto3.common import AWSClientError
+from common.imagebuilder_utils import ImageBuilderStatusMapping
+from pcluster.models.common import BaseTag
+from pcluster.schemas.imagebuilder_schema import ImageBuilderSchema
+from pcluster.templates.cdk_builder import CDKTemplateBuilder
+from pcluster.utils import get_installed_version
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ImageBuilderActionError(Exception):
+    """Represent an error during the execution of an action on the imagebuilder."""
+
+    def __init__(self, message: str, validation_failures: list = None):
+        super().__init__(message)
+        self.validation_failures = validation_failures or []
+
+
+class ImageBuilderStack(Stack):
+    """Class representing a running stack associated to a building image."""
+
+    def __init__(self, name: str):
+        """Init stack info."""
+        super().__init__(name)
+        try:
+            self._image_resource = AWSApi().instance().cfn.describe_stack_resouce(name, "ParallelClusterImage")
+        except AWSClientError:
+            self._image_resource = None
+
+    @property
+    def version(self):
+        """Return the version of ParallelCluster used to create the stack."""
+        return self._get_tag("pcluster_version")
+
+    @property
+    def imagebuild_status(self):
+        """Return the status of the stack of build image process."""
+        cfn_status = self._stack_data.get("StackStatus")
+        for key, value in ImageBuilderStatusMapping.items():
+            if cfn_status in value:
+                return key
+        return None
+
+    @property
+    def image_id(self):
+        """Return output image id."""
+        if self._image_resource:
+            image_build_version_arn = self._image_resource.get("StackResourceDetail").get("PhysicalResourceId")
+            try:
+                resource = AWSApi.instance().imagebuilder.get_image_resources(image_build_version_arn)
+            except AWSClientError:
+                return None
+            image_id = resource.get("image").get("outputResources").get("amis")[0].get("image")
+            return image_id
+        return None
+
+
+class ImageBuilderManager:
+    """Represent a building image, composed by an ImageBuilder config and an ImageBuilderStack."""
+
+    def __init__(self, name: str, config: dict = None, stack: ImageBuilderStack = None):
+        self.name = name
+        self.stack = stack
+        self.template_body = None
+        self.config_version = None
+        self.config = ImageBuilderSchema().load(config)
```

Comment:
Take a look at the already mentioned patch.
I'm saving __source_config and I have self.config as a property, in this way we don't need to load in the initializator, but only when/if using the config.

In this way the loading and validation of syntactic will be done at the same time of the self.config.validate() call, because it first load the config validationg semantically then calls our model validators.

Reply:
good idea, followed